src/PockyBot.NET/Services/Triggers/Peg.cs

```diff
@@ -0,0 +1,77 @@
+using System.Linq;
+using System.Threading.Tasks;
+using GlobalX.ChatBots.Core;
+using GlobalX.ChatBots.Core.Messages;
+using PockyBot.NET.Constants;
+using PockyBot.NET.Persistence.Repositories;
+using PockyBot.NET.Services.Pegs;
+
+namespace PockyBot.NET.Services.Triggers
+{
+    internal class Peg : ITrigger
+    {
+        private readonly IPegRequestValidator _pegRequestValidator;
+        private readonly IPockyUserRepository _pockyUserRepository;
+        private readonly IPegCommentValidator _pegCommentValidator;
+        private readonly IConfigRepository _configRepository;
+        private readonly IChatHelper _chatHelper;
+        private readonly IPegGiver _pegGiver;
+
+        public Peg(IPegRequestValidator pegRequestValidator, IPockyUserRepository pockyUserRepository,
+            IPegCommentValidator pegCommentValidator, IConfigRepository configRepository, IChatHelper chatHelper,
+            IPegGiver pegGiver)
+        {
+            _pegRequestValidator = pegRequestValidator;
+            _pockyUserRepository = pockyUserRepository;
+            _pegCommentValidator = pegCommentValidator;
+            _configRepository = configRepository;
+            _chatHelper = chatHelper;
+            _pegGiver = pegGiver;
+        }
+
+        public string Command => Commands.Peg;
+
+        public bool DirectMessageAllowed => false;
+
+        public bool CanHaveArgs => true;
+
+        public string[] Permissions => new string[0];
```

Comment:
Why string[0]?

Reply:
because multiple permission scopes can be used to access a single piece of functionality. e.g. "ADMIN" and "CONFIG" can both be used to access config

Reply:
Array.Empty<string>()?